File: Publisher-Subscriber/subscriber.py
#!/usr/bin/env python
# -*- coding: UTF-8 -*-
import rospy
from first_package.msg import drone

def func(data):

    rospy.loginfo("received message: (%d,%s,%d,%.2f,%.2f)",
    data.id,data.name,data.speed,data.tempeture,data.battery)

rospy.init_node("subscriber",anonymous= True)

rospy.Subscriber("drone_topic",drone,func)

rospy.spin()

File: Publisher-Subscriber/subscriber.cpp
#include "ros/ros.h"
#include "first_package/zaman.h"

void func(const first_package::zaman::ConstPtr& msg)
{
    ROS_INFO("received message = %d",msg->stamp.sec);
    ROS_INFO("received message = %d",msg->stamp.nsec);
    ROS_INFO("received message = %d",msg->data);

}

int main(int argc, char **argv)
{
    ros::init(argc,argv,"subscriber");

    ros::NodeHandle nh;

    ros::Subscriber first_package_sub = nh.subscribe("msg_time",100,func);

    ros::spin();
    

    return 0;
}
